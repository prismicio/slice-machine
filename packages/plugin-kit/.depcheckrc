ignores:
  [
    "@size-limit/preset-small-lib",
    "@vitest/coverage-v8",
    "monocle-ts",
    "newtype-ts",
    "prettier-plugin-jsdoc",
    "typescript",
  ]
skip - missing: true
