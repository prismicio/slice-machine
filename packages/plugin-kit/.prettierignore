# custom
CHANGELOG.md

# .gitignore copy

/dist
