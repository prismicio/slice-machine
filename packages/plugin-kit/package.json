{
	"name": "@slicemachine/plugin-kit",
	"version": "0.4.27-dev-next-release.0",
	"description": "A set of helpers to develop and run Slice Machine plugins",
	"keywords": [
		"typescript",
		"helpers",
		"utils",
		"toolbox",
		"prismic"
	],
	"repository": {
		"type": "git",
		"url": "ssh://git@github.com/prismicio/slicemachine-plugin-kit.git",
		"directory": "packages/plugin-kit"
	},
	"license": "Apache-2.0",
	"author": "Prismic <contact@prismic.io> (https://prismic.io)",
	"type": "module",
	"exports": {
		".": {
			"types": "./dist/index.d.ts",
			"require": "./dist/index.cjs",
			"import": "./dist/index.js"
		},
		"./fs": {
			"types": "./dist/fs/index.d.ts",
			"require": "./dist/fs/index.cjs",
			"import": "./dist/fs/index.js"
		},
		"./package.json": "./package.json"
	},
	"main": "dist/index.cjs",
	"module": "dist/index.js",
	"types": "dist/index.d.ts",
	"typesVersions": {
		"*": {
			"*": [
				"dist/index.d.ts"
			],
			"fs": [
				"dist/fs/index.d.ts"
			]
		}
	},
	"files": [
		"dist",
		"src"
	],
	"scripts": {
		"build": "vite build",
		"dev": "vite build --watch",
		"format": "prettier --write .",
		"lint": "eslint --max-warnings 0 --ext .js,.ts .",
		"prepack": "npm run build",
		"size": "size-limit",
		"test": "yarn lint && yarn types && yarn unit && yarn build && yarn size",
		"types": "tsc --noEmit",
		"unit": "vitest run --coverage",
		"unit:watch": "vitest watch",
		"depcheck": "depcheck --config=.depcheckrc",
		"audit": "yarn npm audit --environment production --severity high"
	},
	"dependencies": {
		"common-tags": "^1.8.2",
		"defu": "^6.1.2",
		"dotenv": "16.3.1",
		"fp-ts": "^2.13.1",
		"fs-extra": "11.1.1",
		"io-ts": "^2.2.20",
		"io-ts-reporters": "^2.0.1",
		"p-limit": "^4.0.0",
		"prettier": "^3.0.3",
		"prismic-ts-codegen": "^0.1.19"
	},
	"devDependencies": {
		"@prismicio/mock": "0.2.0",
		"@prismicio/types-internal": "2.2.0",
		"@size-limit/preset-small-lib": "8.2.4",
		"@types/common-tags": "1.8.1",
		"@types/fs-extra": "11.0.1",
		"@typescript-eslint/eslint-plugin": "5.55.0",
		"@typescript-eslint/parser": "5.55.0",
		"@vitest/coverage-v8": "0.32.0",
		"depcheck": "1.4.3",
		"eslint": "8.37.0",
		"eslint-config-prettier": "9.0.0",
		"eslint-plugin-prettier": "5.0.1",
		"eslint-plugin-tsdoc": "0.2.17",
		"io-ts-types": "0.5.19",
		"monocle-ts": "2.3.13",
		"newtype-ts": "0.3.5",
		"prettier-plugin-jsdoc": "1.1.1",
		"size-limit": "8.2.4",
		"typescript": "4.9.5",
		"vite": "4.3.9",
		"vite-plugin-sdk": "0.1.1",
		"vitest": "0.32.0"
	},
	"engines": {
		"node": ">=14.15.0"
	}
}
