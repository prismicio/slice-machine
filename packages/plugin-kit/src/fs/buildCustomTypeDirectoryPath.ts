import * as path from "node:path";

import {
	buildCustomTypeLibraryDirectoryPath,
	BuildCustomTypeLibraryDirectoryPathArgs,
} from "./buildCustomTypeLibraryDirectoryPath";

export type BuildCustomTypeDirectoryPathArgs = {
	customTypeID: string;
} & BuildCustomTypeLibraryDirectoryPathArgs;

export const buildCustomTypeDirectoryPath = (
	args: BuildCustomTypeDirectoryPathArgs,
): string => {
	return path.join(
		buildCustomTypeLibraryDirectoryPath(args),
		args.customTypeID,
	);
};
