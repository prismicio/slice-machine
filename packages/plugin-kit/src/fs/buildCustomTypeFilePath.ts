import * as path from "node:path";

import {
	buildCustomTypeDirectoryPath,
	BuildCustomTypeDirectoryPathArgs,
} from "./buildCustomTypeDirectoryPath";

export type BuildCustomTypeFilePathArgs = {
	filename: string;
} & BuildCustomTypeDirectoryPathArgs;

export const buildCustomTypeFilePath = (
	args: BuildCustomTypeFilePathArgs,
): string => {
	return path.join(buildCustomTypeDirectoryPath(args), args.filename);
};
