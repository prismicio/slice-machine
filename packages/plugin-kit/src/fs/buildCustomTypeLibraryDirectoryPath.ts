import { SliceMachineHelpers } from "../createSliceMachineHelpers";

import { CUSTOM_TYPES_LIBRARY_DIRNAME } from "./constants";

export type BuildCustomTypeLibraryDirectoryPathArgs = {
	absolute?: boolean;
	helpers: SliceMachineHelpers;
};

export const buildCustomTypeLibraryDirectoryPath = (
	args: BuildCustomTypeLibraryDirectoryPathArgs,
): string => {
	return args.absolute
		? args.helpers.joinPathFromRoot(CUSTOM_TYPES_LIBRARY_DIRNAME)
		: CUSTOM_TYPES_LIBRARY_DIRNAME;
};
