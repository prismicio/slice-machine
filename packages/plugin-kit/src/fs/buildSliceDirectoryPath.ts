import * as path from "node:path";

import { pascalCase } from "./lib/pascalCase";
import {
	resolveSliceModel,
	ResolveSliceModelArgs,
} from "./lib/resolveSliceModel";

import {
	buildSliceLibraryDirectoryPath,
	BuildSliceLibraryDirectoryPathArgs,
} from "./buildSliceLibraryDirectoryPath";

export type BuildSliceDirectoryPathArgs = BuildSliceLibraryDirectoryPathArgs &
	ResolveSliceModelArgs;

export const buildSliceDirectoryPath = async (
	args: BuildSliceDirectoryPathArgs,
): Promise<string> => {
	const model = await resolveSliceModel(args);

	return path.join(
		buildSliceLibraryDirectoryPath(args),
		pascalCase(model.name),
	);
};
