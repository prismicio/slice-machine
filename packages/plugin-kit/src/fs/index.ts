///////////////////////////////////////////////////////////////////////////////
//
// Project helpers
//
///////////////////////////////////////////////////////////////////////////////

export { checkIsTypeScriptProject } from "./checkIsTypeScriptProject";

export { upsertGlobalTypeScriptTypes } from "./upsertGlobalTypeScriptTypes";
export type { UpsertGlobalTypeScriptTypesArgs } from "./upsertGlobalTypeScriptTypes";

export { checkHasProjectFile } from "./checkHasProjectFile";
export type { CheckHasProjectFileArgs } from "./checkHasProjectFile";

export { writeProjectFile } from "./writeProjectFile";
export type { WriteProjectFileArgs } from "./writeProjectFile";

export { readProjectFile } from "./readProjectFile";
export type { ReadProjectFileArgs } from "./readProjectFile";

export { deleteProjectFile } from "./deleteProjectFile";
export type { DeleteProjectFileArgs } from "./deleteProjectFile";

export { readProjectEnvironment } from "./readProjectEnvironment";
export type {
	ReadProjectEnvironmentArgs,
	ReadEnvironmentReturnType,
} from "./readProjectEnvironment";

export { writeProjectEnvironment } from "./writeProjectEnvironment";
export type { WriteProjectEnvironmentArgs } from "./writeProjectEnvironment";

///////////////////////////////////////////////////////////////////////////////
//
// Slice helpers
//
///////////////////////////////////////////////////////////////////////////////

export { buildSliceDirectoryPath } from "./buildSliceDirectoryPath";
export type { BuildSliceDirectoryPathArgs } from "./buildSliceDirectoryPath";

export { buildSliceFilePath } from "./buildSliceFilePath";
export type { BuildSliceFilePathArgs } from "./buildSliceFilePath";

export { buildSliceLibraryDirectoryPath } from "./buildSliceLibraryDirectoryPath";
export type { BuildSliceLibraryDirectoryPathArgs } from "./buildSliceLibraryDirectoryPath";

export { readSliceLibrary } from "./readSliceLibrary";
export type {
	ReadSliceLibraryArgs,
	ReadSliceLibraryReturnType,
} from "./readSliceLibrary";

export { writeSliceFile } from "./writeSliceFile";
export type { WriteSliceFileArgs } from "./writeSliceFile";

export { writeSliceModel } from "./writeSliceModel";
export type { WriteSliceModelArgs } from "./writeSliceModel";

export { readSliceFile } from "./readSliceFile";
export type { ReadSliceFileArgs } from "./readSliceFile";

export { readSliceModel } from "./readSliceModel";
export type {
	ReadSliceModelArgs,
	ReadSliceModelReturnType,
} from "./readSliceModel";

export { renameSlice } from "./renameSlice";
export type { RenameSliceArgs } from "./renameSlice";

export { deleteSliceFile } from "./deleteSliceFile";
export type { DeleteSliceFileArgs } from "./deleteSliceFile";

export { deleteSliceDirectory } from "./deleteSliceDirectory";
export type { DeleteSliceDirectoryArgs } from "./deleteSliceDirectory";

///////////////////////////////////////////////////////////////////////////////
//
// Custom type helpers
//
///////////////////////////////////////////////////////////////////////////////

export { buildCustomTypeDirectoryPath } from "./buildCustomTypeDirectoryPath";
export type { BuildCustomTypeDirectoryPathArgs } from "./buildCustomTypeDirectoryPath";

export { buildCustomTypeFilePath } from "./buildCustomTypeFilePath";
export type { BuildCustomTypeFilePathArgs } from "./buildCustomTypeFilePath";

export { buildCustomTypeLibraryDirectoryPath } from "./buildCustomTypeLibraryDirectoryPath";
export type { BuildCustomTypeLibraryDirectoryPathArgs } from "./buildCustomTypeLibraryDirectoryPath";

export { readCustomTypeLibrary } from "./readCustomTypeLibrary";
export type {
	ReadCustomTypeLibraryArgs,
	ReadCustomTypeLibraryReturnType,
} from "./readCustomTypeLibrary";

export { writeCustomTypeFile } from "./writeCustomTypeFile";
export type { WriteCustomTypeFileArgs } from "./writeCustomTypeFile";

export { writeCustomTypeModel } from "./writeCustomTypeModel";
export type { WriteCustomTypeModelArgs } from "./writeCustomTypeModel";

export { readCustomTypeFile } from "./readCustomTypeFile";
export type { ReadCustomTypeFileArgs } from "./readCustomTypeFile";

export { readCustomTypeModel } from "./readCustomTypeModel";
export type {
	ReadCustomTypeModelArgs,
	ReadCustomTypeModelReturnType,
} from "./readCustomTypeModel";

export { renameCustomType } from "./renameCustomType";
export type { RenameCustomTypeArgs } from "./renameCustomType";

export { deleteCustomTypeFile } from "./deleteCustomTypeFile";
export type { DeleteCustomTypeFileArgs } from "./deleteCustomTypeFile";

export { deleteCustomTypeDirectory } from "./deleteCustomTypeDirectory";
export type { DeleteCustomTypeDirectoryArgs } from "./deleteCustomTypeDirectory";

export { readSliceTemplateLibrary } from "./readSliceTemplateLibrary";
export type { ReadSliceTemplateLibraryArgs } from "./readSliceTemplateLibrary";
