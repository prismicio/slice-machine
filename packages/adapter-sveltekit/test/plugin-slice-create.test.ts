import { test, expect } from "vitest";
import { createMockFactory } from "@prismicio/mock";
import { createSliceMachinePluginRunner } from "@slicemachine/plugin-kit";
import prettier from "prettier";
import * as fs from "node:fs/promises";
import * as path from "node:path";
import * as tsm from "ts-morph";

import { parseSourceFile } from "./__testutils__/parseSourceFile";
import { testGlobalContentTypes } from "./__testutils__/testGlobalContentTypes";

import adapter from "../src";

/**
 * !!! DO NOT use this mock factory in tests !!!
 *
 * @remarks
 * Its seed is not specific to be used outside the most general cases.
 */
const mock = createMockFactory({ seed: import.meta.url });

// Slice model to be used in general tests.
const model = mock.model.sharedSlice({
	id: "bar_baz",
	name: "QuxQuux",
	variations: [mock.model.sharedSliceVariation()],
});

test("creates a Slice component, model, and global types file on Slice creation", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	expect(
		await fs.readdir(path.join(ctx.project.root, "slices", "QuxQuux")),
	).toStrictEqual(["index.svelte", "model.json"]);
	expect(await fs.readdir(path.join(ctx.project.root, "src"))).toContain(
		"prismicio-types.d.ts",
	);
});

test("supports configuring the location of the global types file", async (ctx) => {
	ctx.project.config.adapter.options.generatedTypesFilePath =
		"./foo/bar/baz.ts";
	const pluginRunner = createSliceMachinePluginRunner({
		project: ctx.project,
		nativePlugins: {
			[ctx.project.config.adapter.resolve]: adapter,
		},
	});
	await pluginRunner.init();

	await pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	expect(
		await fs.readdir(
			path.dirname(
				path.join(
					ctx.project.root,
					ctx.project.config.adapter.options.generatedTypesFilePath,
				),
			),
		),
	).toContain(
		path.basename(ctx.project.config.adapter.options.generatedTypesFilePath),
	);
});

test("upserts a library index.js file", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	expect(await fs.readdir(path.join(ctx.project.root, "slices")))
		.includes("index.js")
		.not.includes("index.ts");
});

test("upserts a library index.ts file when TypeScript is enabled", async (ctx) => {
	ctx.project.config.adapter.options.typescript = true;
	const pluginRunner = createSliceMachinePluginRunner({
		project: ctx.project,
		nativePlugins: {
			[ctx.project.config.adapter.resolve]: adapter,
		},
	});
	await pluginRunner.init();

	await pluginRunner.callHook("slice:create", { libraryID: "slices", model });

	expect(await fs.readdir(path.join(ctx.project.root, "slices")))
		.includes("index.ts")
		.not.includes("index.js");
});

test("library index file includes created Slice", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "index.js"),
		"utf8",
	);
	const file = parseSourceFile(contents);

	expect(
		file
			.getImportDeclarationOrThrow("./QuxQuux/index.svelte")
			.getImportClauseOrThrow()
			.getText(),
	).toBe("QuxQuux");
	expect(
		file
			.getVariableDeclarationOrThrow("components")
			.getInitializerIfKindOrThrow(tsm.SyntaxKind.ObjectLiteralExpression)
			.getPropertyOrThrow("bar_baz")
			.getText(),
	).toBe("bar_baz: QuxQuux");
});

test("model.json file matches the given model", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	expect(
		JSON.parse(
			await fs.readFile(
				path.join(ctx.project.root, "slices", "QuxQuux", "model.json"),
				"utf8",
			),
		),
	).toStrictEqual(model);
});

test("model.json is formatted by default", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "model.json"),
		"utf8",
	);

	expect(contents).toBe(await prettier.format(contents, { parser: "json" }));
});

test("model.json is not formatted if formatting is disabled", async (ctx) => {
	ctx.project.config.adapter.options.format = false;
	const pluginRunner = createSliceMachinePluginRunner({
		project: ctx.project,
		nativePlugins: {
			[ctx.project.config.adapter.resolve]: adapter,
		},
	});
	await pluginRunner.init();

	// Force unusual formatting to detect that formatting did not happen.
	const prettierOptions = { printWidth: 10 };
	await fs.writeFile(
		path.join(ctx.project.root, ".prettierrc"),
		JSON.stringify(prettierOptions),
	);

	await pluginRunner.callHook("slice:create", { libraryID: "slices", model });

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "model.json"),
		"utf8",
	);

	expect(contents).not.toBe(
		await prettier.format(contents, {
			...prettierOptions,
			parser: "json",
		}),
	);
});

test("component file has correct contents", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "index.svelte"),
		"utf8",
	);

	expect(contents).includes("<script>");
	expect(contents).toMatchInlineSnapshot(`
		"<script>
		  /** @type {import(\\"@prismicio/client\\").Content.QuxQuuxSlice} */
		  export let slice;
		</script>

		<section
		  data-slice-type={slice.slice_type}
		  data-slice-variation={slice.variation}
		>
		  Placeholder component for {slice.slice_type} (variation: {slice.variation})
		  Slices
		</section>
		"
	`);
});

test("component file is correctly typed when TypeScript is enabled", async (ctx) => {
	ctx.project.config.adapter.options.typescript = true;
	const pluginRunner = createSliceMachinePluginRunner({
		project: ctx.project,
		nativePlugins: {
			[ctx.project.config.adapter.resolve]: adapter,
		},
	});
	await pluginRunner.init();

	await pluginRunner.callHook("slice:create", { libraryID: "slices", model });

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "index.svelte"),
		"utf8",
	);

	expect(contents).includes('<script lang="ts">');
	expect(contents).toMatchInlineSnapshot(`
		"<script lang=\\"ts\\">
		  import type { Content } from \\"@prismicio/client\\";

		  export let slice: Content.QuxQuuxSlice;
		</script>

		<section
		  data-slice-type={slice.slice_type}
		  data-slice-variation={slice.variation}
		>
		  Placeholder component for {slice.slice_type} (variation: {slice.variation})
		  Slices
		</section>
		"
	`);
});

test("component file is formatted by default", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
	});

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "index.svelte"),
		"utf8",
	);

	expect(contents).toBe(
		await prettier.format(contents, {
			plugins: ["prettier-plugin-svelte"],
			parser: "svelte",
		}),
	);
});

test("component file is not formatted if formatting is disabled", async (ctx) => {
	ctx.project.config.adapter.options.format = false;
	const pluginRunner = createSliceMachinePluginRunner({
		project: ctx.project,
		nativePlugins: {
			[ctx.project.config.adapter.resolve]: adapter,
		},
	});
	await pluginRunner.init();

	// Force unusual formatting to detect that formatting did not happen.
	const prettierOptions = { printWidth: 10 };
	await fs.writeFile(
		path.join(ctx.project.root, ".prettierrc"),
		JSON.stringify(prettierOptions),
	);

	await pluginRunner.callHook("slice:create", { libraryID: "slices", model });

	const contents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "index.svelte"),
		"utf8",
	);

	expect(contents).not.toBe(
		await prettier.format(contents, {
			...prettierOptions,
			plugins: ["prettier-plugin-svelte"],
			parser: "svelte",
		}),
	);
});

testGlobalContentTypes({
	model,
	hookCall: async ({ pluginRunner }) => {
		await pluginRunner.callHook("slice:create", { libraryID: "slices", model });
	},
});

test("component file contains given contents instead of default one", async (ctx) => {
	await ctx.pluginRunner.callHook("slice:create", {
		libraryID: "slices",
		model,
		componentContents: `
			<div>TestSliceCreate</div>
		`,
	});

	const componentContents = await fs.readFile(
		path.join(ctx.project.root, "slices", "QuxQuux", "index.svelte"),
		"utf8",
	);

	expect(componentContents).toContain("<div>TestSliceCreate</div>");
});
