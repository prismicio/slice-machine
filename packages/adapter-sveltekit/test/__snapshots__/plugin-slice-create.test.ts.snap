// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Svelte <=4 syntax > component file has correct contents 1`] = `
"<script>
  /** @type {import(\\"@prismicio/client\\").Content.QuxQuuxSlice} */
  export let slice;
</script>

<section
  data-slice-type={slice.slice_type}
  data-slice-variation={slice.variation}
>
  Placeholder component for {slice.slice_type} (variation: {slice.variation})
  slices.
  <br />
  <strong>You can edit this slice directly in your code editor.</strong>
  <!--
	ðŸ’¡ Use the Prismic MCP server with your code editor
	ðŸ“š Docs: https://prismic.io/docs/ai#code-with-prismics-mcp-server
	-->
</section>
"
`;

exports[`Svelte <=4 syntax > component file is correctly typed when TypeScript is enabled 1`] = `
"<script lang=\\"ts\\">
  import type { Content } from \\"@prismicio/client\\";

  export let slice: Content.QuxQuuxSlice;
</script>

<section
  data-slice-type={slice.slice_type}
  data-slice-variation={slice.variation}
>
  Placeholder component for {slice.slice_type} (variation: {slice.variation})
  slices.
  <br />
  <strong>You can edit this slice directly in your code editor.</strong>
  <!--
	ðŸ’¡ Use the Prismic MCP server with your code editor
	ðŸ“š Docs: https://prismic.io/docs/ai#code-with-prismics-mcp-server
	-->
</section>
"
`;

exports[`component file has correct contents 1`] = `
"<script>
  /* @typedef {import(\\"@prismicio/client\\").Content} Content */
  /* @typedef {import(\\"@prismicio/svelte\\").SliceComponentProps} SliceComponentProps */

  /* @type {SliceComponentProps<Content.QuxQuuxSlice>} */
  const { slice } = $props();
</script>

<section
  data-slice-type={slice.slice_type}
  data-slice-variation={slice.variation}
>
  Placeholder component for {slice.slice_type} (variation: {slice.variation})
  slices.
  <br />
  <strong>You can edit this slice directly in your code editor.</strong>
  <!--
	ðŸ’¡ Use the Prismic MCP server with your code editor
	ðŸ“š Docs: https://prismic.io/docs/ai#code-with-prismics-mcp-server
	-->
</section>
"
`;

exports[`component file is correctly typed when TypeScript is enabled 1`] = `
"<script lang=\\"ts\\">
  import type { Content } from \\"@prismicio/client\\";
  import type { SliceComponentProps } from \\"@prismicio/svelte\\";

  type Props = SliceComponentProps<Content.QuxQuuxSlice>;

  const { slice }: Props = $props();
</script>

<section
  data-slice-type={slice.slice_type}
  data-slice-variation={slice.variation}
>
  Placeholder component for {slice.slice_type} (variation: {slice.variation})
  slices.
  <br />
  <strong>You can edit this slice directly in your code editor.</strong>
  <!--
	ðŸ’¡ Use the Prismic MCP server with your code editor
	ðŸ“š Docs: https://prismic.io/docs/ai#code-with-prismics-mcp-server
	-->
</section>
"
`;
